cpp/unreal_plugins/UrdfRobot/Source/UrdfRobot/UrdfLinkComponent.cpp

```diff
@@ -0,0 +1,175 @@
+            static_mesh_component->bUseDefaultCollision = false;
+            static_mesh_component->SetCollisionResponseToAllChannels(ECollisionResponse::ECR_Ignore);
+        }
+
+        // HACK (MR): Under normal circumstances, neighboring links won't collide with each other because the "Disable Collision" flag is enabled
+        // by default on joints. But this flag has no effect when the "Simulate Physics" option is disabled for a particular link. So, when
+        // "Simulate Physics" is disabled, we need to set the object type to be a special type (e.g., ECollisionChannel::ECC_Vehicle), and then set
+        // all child links to ignore collisions for that type.
+        if (!link_desc->simulate_physics_) {
+            static_mesh_component->bUseDefaultCollision = false;
+            static_mesh_component->SetCollisionObjectType(ECollisionChannel::ECC_Vehicle);
+        }
+
+        if (link_desc->has_parent_ && !link_desc->parent_simulate_physics_) {
+            static_mesh_component->bUseDefaultCollision = false;
+            static_mesh_component->SetCollisionResponseToChannel(ECollisionChannel::ECC_Vehicle, ECollisionResponse::ECR_Ignore);
+        }
+
+        // assign material
+        if (visual_desc.has_material_) {
+
+            // if the material desc is a reference, then follow the pointer to obtain the underlying material desc
+            const UrdfMaterialDesc* material_desc = &(visual_desc.material_desc_); // needs to be const because spear_link_desc is const
+            if (material_desc->is_reference_) {
+                material_desc = material_desc->material_desc_;
+            }
+            SP_ASSERT(material_desc);
+
+            if (material_desc->unreal_material_ != "") {
+                UMaterialInterface* material_interface = LoadObject<UMaterialInterface>(
+                    nullptr, *Unreal::toFString(material_desc->unreal_material_));
+                SP_ASSERT(material_interface);
+                static_mesh_component->SetMaterial(0, material_interface);
+            } else {
+                UMaterialInterface* material_interface = LoadObject<UMaterialInterface>(
+                    nullptr, *Unreal::toFString("/UrdfRobot/Common/Materials/M_PureColor.M_PureColor"));
+                UMaterialInstanceDynamic* material_instance_dynamic = UMaterialInstanceDynamic::Create(material_interface, static_mesh_component);
+                FLinearColor color = {
+                    static_cast<float>(material_desc->color_.at(0)),
+                    static_cast<float>(material_desc->color_.at(1)),
+                    static_cast<float>(material_desc->color_.at(2)),
+                    static_cast<float>(material_desc->color_.at(3))};
+                material_instance_dynamic->SetVectorParameterValue("BaseColor_Color", color);
+                static_mesh_component->SetMaterial(0, material_instance_dynamic);
+            }
+        }
+
+        // connect static mesh component to the top-level link
```

Comment:
Move `SetupAttachment(this)` and `RegisterComponent()` as early as possible, but leave `StaticMeshComponents.Add(static_mesh_component)` at the end.

Reply:
Done